### ISSUE TITLE
SPAN.attachedType in toc should have no width

### ISSUE THREAD

**Issue by AvdN** [nim-lang/Nim#10857]
There is a SPAN with class `attachedType` in each table-of-contents entry in the documentation, that is invisible, but has width. Because of that some entries are wrapped for no apparent reason.

### Example

When looking at the TOC of [osproc](https://nim-lang.org/docs/osproc.html) you'll see:
```
execProcesses
exec
Process
startProcess
```
In this part of the toc, `execProcess` is one entry, and it gets wrapped, although it is shorter than `execProcesses`, because there is a SPAN element on that line with a long CDATA.

### Expected Output
The toc entry should look like:
```
execProcesses
execProcess
startProcess
```

### Possible Solution

The way to solve this is by adding the style `display:none` to the SPAN element. I see three possible ways to do this (presented in order of, what I consider, increased preference):

1. change the `style` attribute in `doc.item.toc` in `config/nimdoc.cfg` to read: `style="visibility:hidden;display:none"`

2. change the `class` attribute in `doc.item.toc` in `config/nimdoc.cfg` to read: `class="attachedType hidden"` (`hidden` is a class that is already in the included style sheet that only sets `"display:none"`)

3. includes the class `attachedType` in the stylesheet section of `doc.file` in `config/nimdoc.cfg` that reads:
```
.attachedType {
  display: none;
  visibility: hidden;
}
```
which makes the the `style` component from `doc.item.toc` superfluous.

### Additional Information

Each of the above solutions will need to be accompanied by changes to the test `nimdoc/testproject/expected/testproject.html` and `nimdoc/testproject/expected/subdir/subdir_b/utils.html`.  

Although the third solution might prevent the need for further changes to `doc.item.toc` if the style further changes, the test files still have to change because of the verbatim inclusion of the stylesheet in the HTML (instead of a LINK to a stylesheet)

----

I can make a PR for this (I wanted to look at generating  the documentation locally anyway (and the test), in order to expiroment with some potential enhancements), but I would like to know if there is any preferential solution, or problem with my preferred solution (the third one), as e.g. I have no idea if there is a reason the style for `attachedType` is currently hard-coded.

**Comment by narimiran** [comment]
First off: thank you for the detailed bug report and for following our template. It makes things much easier for us. (If only more people did that)

When it comes to documentation, I recommend using devel docs available at https://nim-lang.github.io/Nim/lib.html, which have lots of improved details — sometimes the things you see in stable docs will be already fixed there. (But that is not the case this time)

> I have no idea if there is a reason the style for attachedType is currently hard-coded.

I haven't checked it but it could be just some copy-pasted stuff that might be there for no reason :D 

> I can make a PR for this 

Please do (and ping me when it is ready for review). You can go for your third solution, if you think it is the best one.

> Each of the above solutions will need to be accompanied by changes to the test ...

There is a `-d:fixup` flag which you can use which will automatically change those tests for you, e.g.
`nim c -r -d:fixup nimdoc/tester`

**Comment by AvdN** [comment]
@narimiran Since you are subscribed to this issue, I probably don't have to ping you as the PR referenced this issue and added an automatic entry.

I started with `koch test html` but that failed (JsonParsingError) even before I changed anything. `nim c-r nimdoc/tester.nim` worked however, after I made my changes in the config, the same command failed as expected, Then I ran your `nim c -r -d:fixup nimdoc/tester` which also failed, but after that the `nim c-r nimdoc/tester.nim` worked and I committed/created the PR. (You probably know how to do that, the above might help someone (or myself), who comes (back) here looking for clues.)

Thanks
